src/transformer/utf8.hpp

```diff
@@ -17,6 +20,8 @@ uint8_t write_codepoint(uint32_t codepoint, char *utf8_buffer, uint64_t length_l
 uint32_t fetch_next_codepoint(const char *utf8_buffer, uint64_t &position, uint64_t length);
 
 size_t normalize_codepoint(uint32_t codepoint, int32_t *wb_buffer, size_t wb_buffer_length);
+
 bool normalize_string(char **utf8_buffer, uint64_t &buffer_length);
+bool normalize_string(lazy_string &str);
 
 } // namespace ddwaf::utf8
```

Comment:
should this be in src/transformer? there's no transformer class here

Reply:
These are just `utf8` utility functions only used by the transformers so I prefer to keep it here, I can put it in a `transformer/common` directory if it makes things cleaner.